OutOfSchool/OutOfSchool.WebApi/Services/PhotoStorage/Scale.cs

```diff
@@ -0,0 +1,117 @@
+﻿using System.Drawing;
+using System.Drawing.Drawing2D;
+
+namespace OutOfSchool.WebApi.Services.PhotoStorage
+{
+    public class Scale
+    {
+        /// <summary>
+        /// Initializes a new instance of the <see cref="Scale"/> class.
+        /// Scale an image as per given size and keep aspect ratio.
+        /// The final result of the scale may have different width or hight.
+        /// </summary>
+        /// <param name="imgSize">Image size.</param>
+        /// <param name="targetSize">Target size.</param>
+        public Scale(Size imgSize, Size targetSize)
+        {
+            ImgSize = imgSize;
+            Origin = new Point(0, 0);
+
+            TargetSize
+                = targetSize.Height == 0 && targetSize.Width > 0 ? ScaleByWidth(imgSize, targetSize.Width)
+                : targetSize.Width == 0 && targetSize.Height > 0 ? ScaleByHeight(imgSize, targetSize.Height)
+                : AutoScale(imgSize, targetSize);
```

Comment:
1. Looks like a method
2. Does it make any sense to upscale image with 0 width or height ?